### ISSUE TITLE
[Bug]: Can't use shopify cli since updating to v2.11.0 (ruby 3.1.0)

### ISSUE THREAD

**Issue by muriloshimizu** [Shopify/shopify-cli#2031]
### Please confirm that you have:

- [X] Searched [existing issues](https://github.com/Shopify/shopify-cli/issues) to see if your issue is a duplicate. (If you’ve found a duplicate issue, feel free to add additional information in a comment on it.)
- [X] Reproduced the issue in [the latest CLI version](https://github.com/Shopify/shopify-cli/releases).

### What type of project are you working on?

theme

### Expected behavior

Once I updated the shopify cli to version 2.11.0 and ruby to 3.1.0, shopify cli stopped working. I can't log into a shopify store, or run basic commands like `shopify version` anymore.

I suspect this is related to #2027, but I'm not sure.

### Actual behavior

An `✗ An unexpected error occured.` error message is displayed and nothing happens.

### Stack trace

```shell
///////// BREW INSTALL

/usr/local/Cellar/ruby/3.1.0/lib/ruby/3.1.0/open3.rb:222:in `spawn': No such file or directory - node (Errno::ENOENT)
	from /usr/local/Cellar/ruby/3.1.0/lib/ruby/3.1.0/open3.rb:222:in `popen_run'
	from /usr/local/Cellar/ruby/3.1.0/lib/ruby/3.1.0/open3.rb:103:in `popen3'
	from /usr/local/Cellar/ruby/3.1.0/lib/ruby/3.1.0/open3.rb:290:in `capture3'
	from /usr/local/Cellar/shopify-cli/2.11.0/gems/shopify-cli-2.11.0/vendor/deps/cli-kit/lib/cli/kit/system.rb:199:in `delegate_open3'
	from /usr/local/Cellar/shopify-cli/2.11.0/gems/shopify-cli-2.11.0/vendor/deps/cli-kit/lib/cli/kit/system.rb:90:in `capture3'
	from /usr/local/Cellar/shopify-cli/2.11.0/gems/shopify-cli-2.11.0/lib/shopify_cli/context.rb:557:in `capture3'
	from /usr/local/Cellar/shopify-cli/2.11.0/gems/shopify-cli-2.11.0/lib/shopify_cli/environment.rb:17:in `node_version'
	from /usr/local/Cellar/shopify-cli/2.11.0/gems/shopify-cli-2.11.0/lib/shopify_cli/command.rb:104:in `check_node_version'
	from /usr/local/Cellar/shopify-cli/2.11.0/gems/shopify-cli-2.11.0/lib/shopify_cli/command.rb:33:in `call'
	from /usr/local/Cellar/shopify-cli/2.11.0/gems/shopify-cli-2.11.0/lib/shopify_cli/core/executor.rb:17:in `block (2 levels) in call'
	from /usr/local/Cellar/shopify-cli/2.11.0/gems/shopify-cli-2.11.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:37:in `block (2 levels) in with_logging'
	from /usr/local/Cellar/shopify-cli/2.11.0/gems/shopify-cli-2.11.0/vendor/deps/cli-ui/lib/cli/ui/stdout_router.rb:169:in `with_id'
	from /usr/local/Cellar/shopify-cli/2.11.0/gems/shopify-cli-2.11.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:36:in `block in with_logging'
	from /usr/local/Cellar/shopify-cli/2.11.0/gems/shopify-cli-2.11.0/vendor/deps/cli-ui/lib/cli/ui.rb:176:in `log_output_to'
	from /usr/local/Cellar/shopify-cli/2.11.0/gems/shopify-cli-2.11.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:35:in `with_logging'
	from /usr/local/Cellar/shopify-cli/2.11.0/gems/shopify-cli-2.11.0/lib/shopify_cli/core/executor.rb:16:in `block in call'
	from /usr/local/Cellar/shopify-cli/2.11.0/gems/shopify-cli-2.11.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:45:in `block (2 levels) in with_traps'
	from /usr/local/Cellar/shopify-cli/2.11.0/gems/shopify-cli-2.11.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:55:in `twrap'
	from /usr/local/Cellar/shopify-cli/2.11.0/gems/shopify-cli-2.11.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:44:in `block in with_traps'
	from /usr/local/Cellar/shopify-cli/2.11.0/gems/shopify-cli-2.11.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:55:in `twrap'
	from /usr/local/Cellar/shopify-cli/2.11.0/gems/shopify-cli-2.11.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:43:in `with_traps'
	from /usr/local/Cellar/shopify-cli/2.11.0/gems/shopify-cli-2.11.0/lib/shopify_cli/core/executor.rb:15:in `call'
	from /usr/local/Cellar/shopify-cli/2.11.0/gems/shopify-cli-2.11.0/lib/shopify_cli/core/entry_point.rb:24:in `block in call'
	from /usr/local/Cellar/shopify-cli/2.11.0/gems/shopify-cli-2.11.0/lib/shopify_cli/core/monorail.rb:30:in `log'
	from /usr/local/Cellar/shopify-cli/2.11.0/gems/shopify-cli-2.11.0/lib/shopify_cli/core/entry_point.rb:23:in `call'
	from /usr/local/Cellar/shopify-cli/2.11.0/gems/shopify-cli-2.11.0/bin/shopify:37:in `block (2 levels) in <top (required)>'
	from /usr/local/Cellar/shopify-cli/2.11.0/gems/shopify-cli-2.11.0/vendor/deps/cli-kit/lib/cli/kit/error_handler.rb:75:in `handle_abort'
	from /usr/local/Cellar/shopify-cli/2.11.0/gems/shopify-cli-2.11.0/vendor/deps/cli-kit/lib/cli/kit/error_handler.rb:21:in `call'
	from /usr/local/Cellar/shopify-cli/2.11.0/gems/shopify-cli-2.11.0/bin/shopify:36:in `block in <top (required)>'
	from /usr/local/Cellar/shopify-cli/2.11.0/gems/shopify-cli-2.11.0/bin/shopify:47:in `<top (required)>'
	from /usr/local/bin/shopify:7:in `load'
	from /usr/local/bin/shopify:7:in `<main>'


///////// GEM INSTALL

/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/open3.rb:213:in `spawn': No such file or directory - node (Errno::ENOENT)
	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/open3.rb:213:in `popen_run'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/open3.rb:101:in `popen3'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/open3.rb:281:in `capture3'
	from /Library/Ruby/Gems/2.6.0/gems/shopify-cli-2.11.0/vendor/deps/cli-kit/lib/cli/kit/system.rb:199:in `delegate_open3'
	from /Library/Ruby/Gems/2.6.0/gems/shopify-cli-2.11.0/vendor/deps/cli-kit/lib/cli/kit/system.rb:90:in `capture3'
	from /Library/Ruby/Gems/2.6.0/gems/shopify-cli-2.11.0/lib/shopify_cli/context.rb:557:in `capture3'
	from /Library/Ruby/Gems/2.6.0/gems/shopify-cli-2.11.0/lib/shopify_cli/environment.rb:17:in `node_version'
	from /Library/Ruby/Gems/2.6.0/gems/shopify-cli-2.11.0/lib/shopify_cli/command.rb:104:in `check_node_version'
	from /Library/Ruby/Gems/2.6.0/gems/shopify-cli-2.11.0/lib/shopify_cli/command.rb:33:in `call'
	from /Library/Ruby/Gems/2.6.0/gems/shopify-cli-2.11.0/lib/shopify_cli/core/executor.rb:17:in `block (2 levels) in call'
	from /Library/Ruby/Gems/2.6.0/gems/shopify-cli-2.11.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:37:in `block (2 levels) in with_logging'
	from /Library/Ruby/Gems/2.6.0/gems/shopify-cli-2.11.0/vendor/deps/cli-ui/lib/cli/ui/stdout_router.rb:169:in `with_id'
	from /Library/Ruby/Gems/2.6.0/gems/shopify-cli-2.11.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:36:in `block in with_logging'
	from /Library/Ruby/Gems/2.6.0/gems/shopify-cli-2.11.0/vendor/deps/cli-ui/lib/cli/ui.rb:176:in `log_output_to'
	from /Library/Ruby/Gems/2.6.0/gems/shopify-cli-2.11.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:35:in `with_logging'
	from /Library/Ruby/Gems/2.6.0/gems/shopify-cli-2.11.0/lib/shopify_cli/core/executor.rb:16:in `block in call'
	from /Library/Ruby/Gems/2.6.0/gems/shopify-cli-2.11.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:45:in `block (2 levels) in with_traps'
	from /Library/Ruby/Gems/2.6.0/gems/shopify-cli-2.11.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:55:in `twrap'
	from /Library/Ruby/Gems/2.6.0/gems/shopify-cli-2.11.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:44:in `block in with_traps'
	from /Library/Ruby/Gems/2.6.0/gems/shopify-cli-2.11.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:55:in `twrap'
	from /Library/Ruby/Gems/2.6.0/gems/shopify-cli-2.11.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:43:in `with_traps'
	from /Library/Ruby/Gems/2.6.0/gems/shopify-cli-2.11.0/lib/shopify_cli/core/executor.rb:15:in `call'
	from /Library/Ruby/Gems/2.6.0/gems/shopify-cli-2.11.0/lib/shopify_cli/core/entry_point.rb:24:in `block in call'
	from /Library/Ruby/Gems/2.6.0/gems/shopify-cli-2.11.0/lib/shopify_cli/core/monorail.rb:30:in `log'
	from /Library/Ruby/Gems/2.6.0/gems/shopify-cli-2.11.0/lib/shopify_cli/core/entry_point.rb:23:in `call'
	from /Library/Ruby/Gems/2.6.0/gems/shopify-cli-2.11.0/bin/shopify:37:in `block (2 levels) in <top (required)>'
	from /Library/Ruby/Gems/2.6.0/gems/shopify-cli-2.11.0/vendor/deps/cli-kit/lib/cli/kit/error_handler.rb:75:in `handle_abort'
	from /Library/Ruby/Gems/2.6.0/gems/shopify-cli-2.11.0/vendor/deps/cli-kit/lib/cli/kit/error_handler.rb:21:in `call'
	from /Library/Ruby/Gems/2.6.0/gems/shopify-cli-2.11.0/bin/shopify:36:in `block in <top (required)>'
	from /Library/Ruby/Gems/2.6.0/gems/shopify-cli-2.11.0/bin/shopify:47:in `<top (required)>'
	from /usr/local/bin/shopify:23:in `load'
	from /usr/local/bin/shopify:23:in `<main>'
```

### Reproduction steps

1. Install shopify-cli via `gem install shopify-cli` or `brew tap shopify/shopify` and `brew install shopify-cli'
2. Run `shopify version` or `shopify login --store=STORE`

Note: it all started when I ran `brew update` and `brew upgrade`.

### Operating System

macOS Monterey 12.2

### CLI version (run `shopify version` if you're not sure)

2.11.0

### Shell

zsh

### Ruby version (run `ruby -v` if you're not sure)

Tried both 3.1.0 (brew) and 2.6 (default macOS)

### What language and version are you using in your application?

using just for themes

**Comment by amcaplan** [comment]
Hey @muriloshimizu, thanks for the report!

2 things:

1. The error message notes you don't have node installed, and node is required. Others have been experiencing this as well due to some recent changes and we're working on a fix.
2. We don't support Ruby 3.1 just yet; there's also a fix in the works for the brew formula so it'll install 3.0 instead.

**Comment by gonzaloriestra** [comment]
@muriloshimizu the problem was that on 2.11.0 we accidentally required Node for some operations that didn't really need it. It was fixed on 2.11.1, so the theme operations should work now again without nodejs.
